src/components/App/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import logo from '../../logo.svg';
+import './App.css';
+import MainApp from '../MainApp';
+import {
+    useSelector,
+} from 'react-redux';
+
+function App() {
+    const todos = useSelector((state: {list: { todos: any[] }}) => state.list.todos);
```

Comment:
wrong selection just state=>state.list.todos will be fine

Reply:
You should move it to closer components that uses it . Because it rerenders too much